lib/wallet/mod.rs

```diff
@@ -1931,4 +1931,280 @@ impl Wallet {
+                    required_signatures,
+                    parsed_keys
+                        .iter()
+                        .map(|k| k.to_string())
+                        .collect::<Vec<_>>()
+                        .join(",")
+                );
+
+                (address, descriptor)
+            }
+            _ => return Err(miette!("invalid address_type: {}", address_type)),
+        };
+
+        let redeem_script_hex = hex::encode(script.as_bytes());
+
+        Ok((address, redeem_script_hex, descriptor))
+    }
+
+    /// Creates a multisig transaction
+    ///
+    /// This method creates a transaction using UTXOs from a multisig address.
+    /// It returns a PSBT (Partially Signed Bitcoin Transaction) that will need to be signed
+    /// by the required number of keys before it can be broadcast.
+    ///
+    /// # Arguments
+    /// * `multisig_address` - The multisig address to spend from
+    /// * `redeem_script_hex` - The redeem script for the multisig address in hex format
+    /// * `destinations` - Map of destination addresses to amounts
+    /// * `utxos` - List of UTXOs to include in the transaction
+    /// * `fee_policy` - Optional fee policy to use
+    /// * `op_return_message` - Optional OP_RETURN message to include
+    ///
+    /// # Returns
+    /// A tuple containing the PSBT and transaction ID
+    pub fn create_multisig_transaction(
+        &self,
+        multisig_address: &str,
+        redeem_script_hex: &str,
+        destinations: HashMap<bitcoin::Address, Amount>,
+        utxos: Vec<(bitcoin::OutPoint, Amount)>,
+        fee_policy: Option<crate::types::FeePolicy>,
+        op_return_message: Option<Vec<u8>>,
+    ) -> Result<(bitcoin::psbt::Psbt, bitcoin::Txid)> {
+        // Parse the multisig address
+        let validator_network = self.validator().network();
+        let network = bitcoin::Network::from_str(validator_network.to_string().as_str())
+            .map_err(|e| miette!("Failed to convert network: {}", e))?;
+
+        let multisig_addr = bitcoin::Address::from_str(multisig_address)
+            .map_err(|_| miette!("invalid multisig address: {}", multisig_address))?;
+
+        // Verify the address is on the correct network
+        let address = multisig_addr
+            .require_network(network)
+            .map_err(|_| miette!("address network mismatch: expected {}", network))?;
+
+        // Parse the redeem script
+        let redeem_script_bytes = hex::decode(redeem_script_hex)
+            .map_err(|_| miette!("invalid redeem script hex: {}", redeem_script_hex))?;
+
+        let redeem_script = bitcoin::ScriptBuf::from_bytes(redeem_script_bytes);
+
+        // Create a new transaction
+        let mut tx = bitcoin::Transaction {
+            version: bitcoin::transaction::Version::TWO,
+            lock_time: bitcoin::absolute::LockTime::ZERO,
+            input: Vec::new(),
+            output: Vec::new(),
+        };
```

Comment:
I'm very skeptical towards building the transaction this way, I have to say. What about the BDK transaction builder?

Reply:
Without using the transaction builder, there's no point in this function being in the wallet, either